projects/libs/core-library-typescript/src/virtual-machine/virtual-machine.ts

```diff
@@ -29,9 +29,13 @@ export class VirtualMachine {
         }, transactionScopeStateCache);
 
       } catch (err) {
-        logger.error(`transaction ${JSON.stringify(transaction)} failed. error: ${err}`);
-        rejectedTransactions.push(transaction);
-        continue;
+        if (!err.expected) {
+          throw err;
+        } else {
```

Comment:
You don't have to have the `else` here, since `throw err` exists the context.

Reply:
I prefer this style. I like code branches in the same depths to have the same indentation.

Reply:
Alrighty.